plugin.video.orftvthek/resources/lib/htmlscraper.py

```diff
@@ -0,0 +1,1040 @@
+			article = parseDOM(item, name='article', attrs={'class': 'b-livestream-teaser is-live.*?'})
+			if not len(bundesland_article) and len(article):
+				figure = parseDOM(article, name='figure', attrs={'class': 'teaser-img'}, ret=False)
+				image = parseDOM(figure, name='img', attrs={}, ret='data-src')
+				image = replaceHTMLCodes(image[0])
+
+				time = parseDOM(article, name='h4', attrs={'class': 'time'}, ret=False)
+				time = replaceHTMLCodes(time[0])
+				time = stripTags(time)
+
+				title = parseDOM(article, name='h4', attrs={'class': 'livestream-title.*?'})
+				title = replaceHTMLCodes(title[0])
+
+				link = parseDOM(item, name='a', attrs={'class': 'js-link-box'}, ret="href")
+				link = replaceHTMLCodes(link[0])
+
+				online = parseDOM(article, name='span', attrs={'class': 'status-online'})
+				if len(online):
+					online = True
+				else:
+					online = False
+
+				restart = parseDOM(article, name='span', attrs={'class': 'is-restartable'})
+				if len(restart):
+					restart = True
+				else:
+					restart = False
+
+				self.buildLivestream(title, link, time, restart, channel, image, online)
+			elif len(bundesland_article):
+				bundesland_data = replaceHTMLCodes(bundesland_article[0])
+				bundesland_data = json.loads(bundesland_data)
+				for bundesland_item_key in bundesland_data:
+					bundesland_item = bundesland_data.get(bundesland_item_key)
+					if bundesland_item and bundesland_item is not True and len(bundesland_item):
+						bundesland_title = bundesland_item.get('title')
+						bundesland_image = bundesland_item.get('img')
+						bundesland_link = bundesland_item.get('url')
+
+						self.buildLivestream(bundesland_title, bundesland_link, "", True, channel, bundesland_image, True)
+
+	def buildLivestream(self, title, link, time, restart, channel, banner, online):
+		html = fetchPage({'link': link})
+		container = parseDOM(html.get("content"), name='div', attrs={'class': "player_viewport.*?"})
+		if len(container):
+			data = parseDOM(container[0], name='div', attrs={}, ret="data-jsb")
+
+			if online:
+				state = (self.translation(30019))
+			else:
+				state = (self.translation(30020))
+
+			if time:
+				time_str = " (%s)" % time
+			else:
+				time_str = ""
+
+			try:
+				xbmcaddon.Addon('inputstream.adaptive')
+				inputstreamAdaptive = True
+			except RuntimeError:
+				inputstreamAdaptive = False
+
+			if channel:
+				channel = "[%s]" % channel
+			else:
+				channel = "LIVE"
+
+			uhd_streaming_url = self.getLivestreamUrl(data, 'uhdbrowser')
+			if uhd_streaming_url:
+				debugLog("Adding UHD Livestream")
+				uhdContextMenuItems = []
+				if inputstreamAdaptive and restart and online:
+					uhdContextMenuItems.append(('Restart', 'RunPlugin(plugin://%s/?mode=liveStreamRestart&link=%s)' % (
+						xbmcaddon.Addon().getAddonInfo('id'), link)))
+					uhd_final_title = "[Restart]%s[UHD] - %s%s" % (channel, title, time_str)
```

Comment:
this needs to be localized

Reply:
done. although the translation for the string is the same in german. the feature on the website/app is also called "Restart" in german.